### PR TITLE
Some coverity errors and clang tidy replacements

### PR DESCRIPTION
**Pull request by mir3x**

### CODE REVIEW

**Comment by lmoureaux** [comment]
Can you drop these:
image
And don't forget to mark as ready when you're done.

File: utility/cmake_fc_config.h.in [hunk 1]
```diff
@@ -57,8 +57,6 @@
 /* memory.h available */
 #cmakedefine HAVE_MEMORY_H
 
-/* pwd.h available */
-#cmakedefine HAVE_PWD_H
```
Comment (lmoureaux) [inline utility/cmake_fc_config.h.in:61]: There must be `HAVE_GETPWUID` somewhere, you removed it from `FindDependencies` so it can be dropped too.

**Review by lmoureaux — COMMENTED** [review]
I wonder how many char * are left now :D

Can you squash 0c8ddd4d51d1b9bfcbc70c5353f2ab3202cfba6d and 5b0a1deac748d759005a1d5f124a7bb1ac3ce69c together to have a cleaner history in lua?

File: utility/shared.cpp [hunk 2]
```diff
@@ -1534,9 +1534,9 @@ void free_multicast_group() { NFCNPP_FREE(mc_group); }
  ****************************************************************************/
 void interpret_tilde(char *buf, size_t buf_size, QString filename)
 {
-  if (filename.startsWith("~/")) {
+  if (filename.startsWith(QString("~") + QString(DIR_SEPARATOR_CHAR))) {
```
[the remark this answers is not in the document]
Comment (mir3x) [reply utility/shared.cpp:1492]: it doeesnt work that way, DIR_SEPARATOR-CHAR  .. is QDir()::separator()
I replaced it to "~/" bc it replaces home directory only and nothing else.

**Pull request closed without merging**